delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/SseEventLog.scala

```diff
@@ -0,0 +1,95 @@
+package ch.epfl.bluebrain.nexus.delta.sdk
+
+import akka.persistence.query.Offset
+import cats.syntax.all._
+import ch.epfl.bluebrain.nexus.delta.kernel.Mapper
+import ch.epfl.bluebrain.nexus.delta.sdk.model._
+import ch.epfl.bluebrain.nexus.delta.sdk.model.organizations.OrganizationRejection
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.{ProjectRef, ProjectRejection}
+import ch.epfl.bluebrain.nexus.delta.sourcing.EventLog
+import fs2.Stream
+import monix.bio.{IO, Task}
+
+/**
+  * An event log that reads events from a [[Stream]] and transforms each event to JSON-LD in preparation for consumption by SSE routes
+  */
+trait SseEventLog {
+
+  /**
+    * Get stream of all events as ''T''.
+    *
+    * @param offset the offset to start from
+    */
+  def stream(offset: Offset): Stream[Task, Envelope[JsonLdValue]]
+
+  /**
+    * Get stream of events inside an organization as ''T'', transforming the error from ''OrganizationRejection'' to ''R''.
+    *
+    * @param org    the organization label
+    * @param offset the offset to start from
+    */
+  def stream[R](
+      org: Label,
+      offset: Offset
+  )(implicit mapper: Mapper[OrganizationRejection, R]): IO[R, Stream[Task, Envelope[JsonLdValue]]]
+
+  /**
+    * Get stream of events inside a project as ''T'', transforming the error from ''ProjectRejection'' to ''R''.
+    *
+    * @param project the project reference
+    * @param offset  the offset to start from
+    */
+  def stream[R](
+      project: ProjectRef,
+      offset: Offset
+  )(implicit mapper: Mapper[ProjectRejection, R]): IO[R, Stream[Task, Envelope[JsonLdValue]]]
+}
+
+object SseEventLog {
+
+  /**
+    * An event log that reads events from a [[Stream]] and transforms each event to JSON-LD that is available through ''exchanges''.
+    * The JSON-LD events are then used for consumption by SSE routes
+    */
+  def apply(
+      eventLog: EventLog[Envelope[Event]],
+      orgs: Organizations,
+      projects: Projects,
+      exchanges: Set[EventExchange]
+  ): SseEventLog = new SseEventLog {
+
+    private lazy val exchangesList = exchanges.toList
+
+    def stream(offset: Offset): Stream[Task, Envelope[JsonLdValue]] =
```

Comment:
We could also serialize the enveloppe too no and return the complete serialized value no ?
`Envelope[JsonLdValue]]` seems a little weird, either we want the `Envelope[Event]` or the JsonLd no ?

Reply:
No, we want both, asi you can see in `ResponseToJsonLd`.

Reply:
And I don't want to change `ResponseToJsonLd`  even more to accommodate to this.
`Envelope[JsonLdValue]` is fine since the `JsonLdValue` still contains the `Event`. So you keep having all the information if you wanted

Reply:
We have to send a `ServerSentEvent`, which requires the json data plus some of the envelope fields.
I could make the `SseEventLog` return the `ServerSentEvent` directly. But I don't think that's any better because
1. Is akka specific.
2. We will still need `Envelope` to support all the other use-cases in `ResponseToJsonLd `

So in order to keep `ResponseToJsonLd` sane and to be aligned with the regular SSE events, I think it is a good tradeoff

Reply:
We can see it later but maybe the regular events could evolve a bit to have a common ground for the two types of SSEs

Reply:
I'm open to something like that If it makes things more readable. So let's consider that for a separate PR if we arrive to a better solution.

Reply:
I am ok with it, so let's open issues for the different points to resolve in other PRs so that we talk further about it during the next issue curation session.